### ISSUE TITLE
My Server needs a look from a trained eye QwQ

### ISSUE THREAD

**Issue by notIkibo** [lucko/spark#341]
Hihi dear lucko, 

I am currently running my own little server with my friends which is quite cool, but eventually the server lags. Even with 2 people the TPS goes down to like 13. I really want to solve this, but hardware is totally fine. apparently there is something wrong with my world, but I can't really see anything. So I am sending you my profilers. If anyone has time: Please tell me whats wrong, I would like to know how you came to your conclusion and what I can do to fix it.

I even asked the support, if they throttle their cpu perfomance. But anyway, I want to live without lag qwq

https://www.dropbox.com/sh/4rn9unwz4rp9sr1/AADpiPLPff3dKFvo_FT6RS7va?dl=0

**Comment by woodfairy** [comment]
<img width="317" alt="Bildschirmfoto 2023-07-01 um 11 41 46" src="https://github.com/lucko/spark/assets/66573402/c018f469-599b-4c54-9f35-45aed8bd2f16">

Your server has a lot of memory pressure, this might be the cause

**Comment by notIkibo** [comment]
(I think someone took a look answered my issue here, but deleted it afterwards. Though still thank you :3)

I am no where near knowing the cause, but what my entity count rises and rises. Especially item entities (Like drops of mobs?). I once did /kill @e which killed like 1400 entities (but apparently it can pile up to 3800) and the lag was g o n e. (My villagers too). My players have no farm that leaves something on the ground btw. 

So I would be open for every tip on entity control. Maybe even how to limit on item entities? Is that possible?

I asked someone on the Discord sever about my lag and they were seeing that there was a lot of information exchange around entities and suspected a datapack of mine. Though I don't really have datapack that meddles with entities. I think maybe Vanilla Refresh when you look at mobs and see their health etc., but I turned it off since I have Jade (WTHIT) and now many other things as a counter measure.

In a nutshell, item entities pile up somewhere somehow and cause lag. I don't know the cause, but trying to control it wouldn't hurt though I don't know any mods for Fabric 1.19.2 that can do that. And datapacks can apparently be the cause, but Im not convinced.

Sincerely
Ikibo

PS: As much fun it is to create your own server: I hate damage control. It's no fun to play with four people and it already lags while the problem is invisible. And the hardware is chillin on around 30% use. Like how does it still lag, if it's not on a 100%?

**Comment by woodfairy** [comment]
It was me that answered! I thought you had severe memory pressure bc I saw you physical memory being almost exhausted. Well turned out be GB not MB :P I don't know the cause of your issues but I also noticed that you have many entities. What type of server are you running? I switched from Paper to Pufferfish and I don't have any performance issues anymore (similar hardware specs). Is there something near spawn that could cause this? Is it maybe a plugin causing this? You can do some kind configuration for entity behavior, but depends on your server type

You could try to log entity related function calls, look what kind of entities are causing the issues specifically, and try to find the root cause. One possible option: Try every plugin/datapack by disabling them all and then re-enabling them step by step, it takes time, but if a plugin is the cause you can find it this way.
